rules/prefer-string-starts-ends-with.js

```diff
@@ -64,33 +70,62 @@ const create = context => {
 				return;
 			}
 
+			function fix(fixer, {useNullishCoalescing, useOptionalChaining, useStringCasting} = {}) {
+				const method = result.messageId === MESSAGE_STARTS_WITH ? 'startsWith' : 'endsWith';
+				const [target] = node.arguments;
+				let targetString = sourceCode.getText(target);
+				const isRegexParenthesized = isParenthesized(regexNode, sourceCode);
+				const isTargetParenthesized = isParenthesized(target, sourceCode);
+
+				if (
+					// If regex is parenthesized, we can use it, so we don't need add again
+					!isRegexParenthesized &&
+					(isTargetParenthesized || shouldAddParenthesesToMemberExpressionObject(target, sourceCode))
```

Comment:
And this need change, because the MemberExpression.object is not the old `target` any more, for example.

```js
/^a/.test(1)
```


```js
(1).startsWith()
```
this need add `()`

```js
String(1).startsWith()
```

this don't need.

Reply:
I think this comment is outdated.